src/lib.rs

```diff
@@ -505,6 +534,70 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight(1000_000)] // TODO weights
+		pub fn clear_expired_match(
+			origin: OriginFor<T>,
+			match_id: T::Hash
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let chess_match = match Self::chess_matches(match_id) {
+				Some(m) => m,
+				None => return Err(Error::<T>::NonExistentMatch.into()),
+			};
+
+			ensure!(
+				(chess_match.state == MatchState::OnGoing(NextMove::Whites)) | 
+				(chess_match.state == MatchState::OnGoing(NextMove::Blacks)), 
+				Error::<T>::MatchNotOnGoing
+			);
+			
+			let now = <frame_system::Pallet<T>>::block_number();
+			let diff = now - chess_match.last_move;
+			let expired: bool = match chess_match.style {
+				MatchStyle::Bullet => diff > T::BulletPeriod::get(),
+				MatchStyle::Blitz => diff > T::BlitzPeriod::get(),
+				MatchStyle::Rapid => diff > T::RapidPeriod::get(),
+				MatchStyle::Daily => diff > T::DailyPeriod::get(),
+			};
+
+			ensure!(expired, Error::<T>::MatchNotExpired);
+
+			let winner = match chess_match.state {
+				MatchState::OnGoing(NextMove::Whites) => chess_match.opponent.clone(),
+				_ => chess_match.challenger.clone(),
+			};
+
+			Self::deposit_event(Event::MatchWon(match_id, winner.clone(), chess_match.board.clone()));
+
+			let time_since_expiry = match chess_match.style {
+				MatchStyle::Bullet => now - (chess_match.last_move + T::BulletPeriod::get()),
+				MatchStyle::Blitz => now - (chess_match.last_move + T::BlitzPeriod::get()),
+				MatchStyle::Rapid => now - (chess_match.last_move + T::RapidPeriod::get()),
+				MatchStyle::Daily => now - (chess_match.last_move + T::DailyPeriod::get()),
+			};
+
+			if (who == chess_match.challenger) | (who == chess_match.opponent) | (time_since_expiry <= T::IncentivePeriod::get()) {
+				// winner gets both deposits
+				match chess_match.win_bet(&winner) {
+					Ok(()) => {},
+					Err(_) => Self::deposit_event(Event::MatchAwardError(match_id, winner)),
+				}
+			} else {
```

Comment:
```suggestion
			} else if who != loser {
```

if we allow the loser to do this, they could use this as a strategy in the game

they refuse to play because they think there're about to lose anyway
then they try to call `clear_expired_bet` to get part of their bet back

there's many factors that make this attack vector very narrow (e.g.: winner can just claim their victory within the incentive period)

but still, loser should always lose 100% of their bet... players should never have incentives to attack their opponent by refusing to play

Reply:
Actually, the conditions for getting a part of the bet (the incentive) is only if the caller is not the challenger, nor the opponent :) if either of them calls the function, the winner gets full both bets: 

```
if (who == chess_match.challenger) | (who == chess_match.opponent) | (time_since_expiry <= T::IncentivePeriod::get()) {
				// winner gets both deposits
				match chess_match.win_bet(&winner) {
					Ok(()) => {},
					Err(_) => Self::deposit_event(Event::MatchAwardError(match_id, winner)),
				}
			} else { ... }
```

Reply:
that's correct, I'll cancel the suggestion